state-chain/pallets/cf-trading-strategy/src/lib.rs

```diff
@@ -50,6 +50,8 @@ pub const PALLET_VERSION: StorageVersion = StorageVersion::new(1);
 // have a fixed order id (at least until we develop more advanced strategies).
 const STRATEGY_ORDER_ID: OrderId = 0;
 
+impl_pallet_safe_mode!(PalletSafeMode; trading_strategies_enabled);
```

Comment:
probably makes sense to have this more granular, so add / deploy and close are separate options in safe mode.

Reply:
I agree I think it makes sense to differentiate between: 
- add/update
- close
- execution (`on_idle`)

I can see why we might want to pause updates, but allow closing of strategies (in fact it might be nice to be able to force-close all strategies, but that's another topic). An I can also see why we might want to allow strategy updates but temporarily pause order updates.

Reply:
OK, split into three options:
- strategy_updates_enabled
- strategy_closure_enabled
- strategy_execution_enabled